docs/source/dev-guide/plugin-api/subcommand_guide.rst

```diff
@@ -0,0 +1,179 @@
+=================================
+Custom subcommand plugin tutorial
+=================================
+
+In this tutorial, we will create a new ``conda`` subcommand that can convert a string
+into ASCII art.
+
+To follow along with this guide, it is recommended that you create and activate a new ``conda``
+environment with the following commands:
+
+.. code-block:: bash
+
+   $ conda create -n plugin-tutorial "python>=3"
+
+   $ conda activate plugin-tutorial
+
+
+Project directory structure
+---------------------------
+
+Set up your working directory and files as shown below:
+
+.. code-block:: bash
+
+    string-art
+    │── string_art.py
+    └── setup.py
+
+
+The custom subcommand module
+----------------------------
+
+The following module implements a function, ``conda_string_art`` (where a specified string gets
+converted into ASCII art), into a plugin manager hook called ``conda_subcommands``.
+
+The ``HookImplMarker`` decorator is initialized with the name of ``conda`` as the host
+project in the ``conda/plugins/__init__.py`` file, and it is invoked via ``@conda.plugins.register``
+in the example subcommand module below:
+
+.. code-block:: python
+   :caption: string-art/string_art.py
+
+   from art import text2art
+   from typing import Sequence
+
+
+   import conda.plugins
+
+
+   def conda_string_art(args: Sequence[str]):
+       # if using a multi-word string with spaces, make sure to wrap it in quote marks
+       output = "".join(args)
+       string_art = text2art(output)
+
+       print(string_art)
+
+
+   @conda.plugins.register
+   def conda_subcommands():
+       yield conda.plugins.CondaSubcommand(
+           name="string-art",
+           summary="tutorial subcommand that prints a string as ASCII art",
+           action=conda_string_art,
+       )
+
+
+Entrypoint namespace for the custom subcommand
+----------------------------------------------
+
+In order to run the ``conda string-art`` subcommand successfully, you will need to make sure
+that the ``art`` package is available, which is why it is listed in the ``install_requires``
+section of the ``setup.py`` file shown below:
+
+.. code-block:: python
+   :caption: string-art/setup.py
+
+   from setuptools import setup
+
+   install_requires = [
+       "conda",
+       "art",
+   ]
+
+   setup(
+       name="my-conda-subcommand",
+       install_requires=install_requires,
+       entry_points={"conda": ["my-conda-subcommand = string_art"]},
+       py_modules=["string_art"],
+   )
+
```

Comment:
I recently ran through setting up my own plugin and got stuck on configuring this ``setup`` function call correctly. It's a little confusing as I wasn't completely aware of how ``setup.py`` files are configured (been a while since I've had to set them up 😅 ). Below are my suggestions for adding a little more detail to this section and hopefully making it easier to those reading it, especially people who may not be super familiar with ``setup.py``.

```suggestion

Here is a further explanation of the variables that we are passing to the ``setup`` function:

**name** This is the name the package which contains your subcommand. This is also how others will find your subcommand package if you choose to upload it to PyPi.

**install_requires** These are all the dependencies for your project. This should at a minimum always contain the version of ``conda`` for which your plugin is compatible with.

**entry_points** The entry point you list here is how ``conda`` will discover your plugin and should point to the file containing the ``conda.plugins.register`` hook. In our simple use case, it points to the ``string_art`` module contained within the ``string_art.py`` file. For more complex examples where your module is contained within a folder, it may look like this, ``my_module.main`` or ``my_modules.plugin_hooks``.

 **py_modules** The ``py_modules`` variables lets ``setup`` know exactly where to look for all the modules which comprise your plugin source code. 

```

Reply:
I made a few wording changes but I went ahead and added this, thank you for the suggestion!